### ISSUE TITLE
Check.assert.array.of.functions with custom Error message

### ISSUE THREAD

**Issue by dkhuntrods** [philbooth/check-types.js#75]
Hi! First thanks for this great package, really useful.

I'm just starting out with it, so please forgive if I've made any incorrect assumptions, but I have the impression from the docs that calling `check.assert.array.of.function` with a custom error message ought to work:
```
check.assert.array.of.function(arrayOfFunctions, 'Error message');
// Throws 'TypeError: Invalid function'
```

If I set a custom Type, it accepts that, though not in the place I would've expected:
```
check.assert.array.of.function(arrayOfFunctions, CustomError);
// Throws 'CustomError: Invalid function'
```

What  I'd expect is:
```
check.assert.array.of.function(arrayOfFunctions, 'Error message');
// Throws 'TypeError: Error message'
check.assert.array.of.function(arrayOfFunctions, 'Error message', CustomError);
// Throws 'CustomError: Error message'
```

Is this expected, and if so, is there an simple way to achieve what I'd like to (without having to loop  through the array)?

Thanks!

**Comment by philbooth** [comment]
No, that looks like a bug, sorry. I'll dig into it properly later on!

**Comment by philbooth** [comment]
Apologies again for this, it should be fixed in version `7.1.3`, published just now. Let me know if you get any problems!
